### PR TITLE
Set ForceNew on the path field of namespaces

### PR DESCRIPTION
**Pull request by kschoche**
With Terraform Vault provider, the current behaviour is that changing "path" argument of "vault_namespace" resource creates a new namespace but does not destroy the old one, nor update the existing namespace.

### CODE REVIEW

File: vault/resource_namespace_test.go [hunk 1]
```diff
@@ -45,6 +45,15 @@ func TestAccNamespace(t *testing.T) {
...
 		CheckDestroy: testNamespaceDestroy(namespacePath),
 		Steps: []resource.TestStep{
+			{
+				Config: testNamespaceConfig(namespacePath),
+				Check:  testNamespaceCheckAttrs(),
+			},
+			{
+				Config: testNamespaceConfig(namespacePath + "-foo"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("vault_namespace.test", consts.FieldPath, namespacePath+"-foo")),
```
Comment (benashz) [inline vault/resource_namespace_test.go:55]: I wonder if we should also check to that the old namespace was properly deleted in Vault? You can do that in another `TestStep`? I think `testNamespaceDestroy()` might work here.

**Review by benashz — COMMENTED** [review]
Looking great! Just had a comment about checking that the old namespace was properly deleted.

File: vault/resource_namespace_test.go (hunk 1 above)
Comment (kschoche) [reply vault/resource_namespace_test.go:55]: Great idea, I was actually thinking about adding that, and so will do so!

Comment (kschoche) [reply vault/resource_namespace_test.go]: I added this using `testNamespaceDestroy()` as an additional `Check` as part of the second step and it looks like it works to check that the original namespace with that path was destroyed. I wasnt sure what to use for the `Config` portion of an additional `TestStep` yet, but if you have ideas I'd love to chat offline!

**Review by benashz — APPROVED** [review]
LGTM!

**Pull request merged**